main.go

```diff
@@ -34,6 +34,35 @@ func main() {
 		version.PrintVersionAndExit()
 	}
 
+	if s.AutoDiscoverBaseArn {
+		if s.BaseRoleARN != "" {
+			log.Fatal("--auto-discover-base-arn can't be used if --base-role-arn is specified")
+		}
+		if s.DefaultIAMRole == "" {
+			log.Fatal("You need to specify the default IAM role, use --auto-discover-default-role to autodiscover it")
```

Comment:
Well even if you set `--auto-discover-default-role` it won't work as the test is only if the default role is set which will not have happen at this stage yet. Anyhow see my other comment about why this is even mandatory (maybe I'm missing something :)?).